MiniGamesBox Classic/src/main/java/plugily/projects/minigamesbox/classic/kits/KitRegistry.java

```diff
@@ -56,15 +59,59 @@ public void registerKit(Kit kit) {
       plugin.getDebugger().debug("Kit " + kit.getKey() + " can't be added as its already registered");
       return;
     }
-    FileConfiguration config = ConfigUtils.getConfig(plugin, "kits");
-    if(!config.getBoolean("Enabled-Game-Kits." + kit.getKey(), false)) {
+
+    initializeKitConfig(kit);
+    kit.saveKitsConfig();
+    plugin.getDebugger().debug("Kit " + kit.getKey() + "'s config files have been initialized");
+
+    if(!configurationSection.getBoolean("enabled", false)) {
       plugin.getDebugger().debug("Kit " + kit.getKey() + " is disabled by kits.yml");
       return;
     }
+
+    loadKitConfig(kit);
+    plugin.getDebugger().debug("Kit " + kit.getKey() + "'s config files have been loaded");
+
     plugin.getDebugger().debug("Registered {0} kit", kit.getKey());
     kits.add(kit);
   }
 
+  public void initializeKitConfig(Kit kit) {
+    ConfigurationSection configurationSection = kit.getKitsConfig().getConfigurationSection(kit.getKey());
+    if (configurationSection == null) {
+      configurationSection = kit.getKitsConfig().createSection(kit.getKey());
+    }
+    configurationSection.set("name", kit.getName());
+    if (!configurationSection.contains("enabled")) {
+      configurationSection.set("enabled", true);
+    }
+    int currentItem = 0;
+    if (!configurationSection.contains("items")) {
+      ConfigurationSection finalConfigurationSection = configurationSection;
+      kit.getKitItems().forEach((item, indexes) -> {
+        ConfigurationSection itemConfigurationSection = finalConfigurationSection.createSection(String.valueOf(currentItem));
+        XItemStack.serialize(item, itemConfigurationSection.createSection("item"));
+        itemConfigurationSection.set("slots", indexes);
+      });
+    }
+  }
+
+  public void loadKitConfig(Kit kit) {
+    ConfigurationSection configurationSection = kit.getKitsConfig().getConfigurationSection(kit.getKey());
+    assert configurationSection != null;
+    kit.setName(configurationSection.getString("name"));
+
+    kit.getKitItems().clear();
+    HashMap<ItemStack, List<Integer>> kitItems = new HashMap<>();
+
+    Objects.requireNonNull(configurationSection.getConfigurationSection("items"), "Items for kit " + kit.getKey() + " is null").getKeys(false).forEach((k) -> {
```

Comment:
Consider using if-null condition rather than fail-fast exception. Users won't like exceptions.

Reply:
Resolved, thanks for pointing out.